atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/CommitTsLoader.java

```diff
@@ -49,12 +52,22 @@ private CommitTsLoader(TLongLongMap commitTsByStartTs, TransactionService transa
     public static CommitTsLoader create(TransactionService transactionService, TLongSet startTssToWarmingCache) {
         TLongLongMap cache = new TLongLongHashMap();
         if (!startTssToWarmingCache.isEmpty()) {
-            // Ideally TransactionService should work with primitive collections to avoid GC overhead..
-            cache.putAll(transactionService.get(TDecorators.wrap(startTssToWarmingCache)));
+            warmCache(transactionService, startTssToWarmingCache, cache);
+
         }
         return new CommitTsLoader(cache, transactionService);
     }
 
+    private static void warmCache(
```

Comment:
I'd like to get rid of this and try to reuse the cache instead. But outside the scope of this PR.